src/python/pants/backend/python/target_types.py

```diff
@@ -258,6 +282,7 @@ def spec(self) -> str:
         return self.name
 
 
+# @TODO: Remove the SecondaryOwnerMixin in Pants 2.14.0
```

Comment:
I don't think this can be removed until `use_deprecated_pex_binary_run_semantics` is removed. Unless the idea is that the dedicated deprecation you have for file arguments covers this?